In [134]:
#Importing important libraries
import pandas as pd
import numpy as np

In [135]:
#Mouting dataset file
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


#STEP 1 :- Data Understanding

In [136]:
#Reading datset in csv format.
df = pd.read_csv("/content/drive/MyDrive/student-dropout/data.csv",sep=";")

#Getting first five rows
df.head()

Marital status  Application mode  Application order  Course  \
0               1                17                  5     171   
1               1                15                  1    9254   
2               1                 1                  5    9070   
3               1                17                  2    9773   
4               2                39                  1    8014   

   Daytime/evening attendance\t  Previous qualification  \
0                             1                       1   
1                             1                       1   
2                             1                       1   
3                             1                       1   
4                             0                       1   

   Previous qualification (grade)  Nacionality  Mother's qualification  \
0                           122.0            1                      19   
1                           160.0            1                       1   
2                           122.0            1                      37   
3                           122.0            1                      38   
4                           100.0            1                      37   

   Father's qualification  ...  Curricular units 2nd sem (credited)  \
0                      12  ...                                    0   
1                       3  ...                                    0   
2                      37  ...                                    0   
3                      37  ...                                    0   
4                      38  ...                                    0   

   Curricular units 2nd sem (enrolled)  \
0                                    0   
1                                    6   
2                                    6   
3                                    6   
4                                    6   

   Curricular units 2nd sem (evaluations)  \
0                                       0   
1                                       6   
2                                       0   
3                                      10   
4                                       6   

   Curricular units 2nd sem (approved)  Curricular units 2nd sem (grade)  \
0                                    0                          0.000000   
1                                    6                         13.666667   
2                                    0                          0.000000   
3                                    5                         12.400000   
4                                    6                         13.000000   

   Curricular units 2nd sem (without evaluations)  Unemployment rate  \
0                                               0               10.8   
1                                               0               13.9   
2                                               0               10.8   
3                                               0                9.4   
4                                               0               13.9   

   Inflation rate   GDP    Target  
0             1.4  1.74   Dropout  
1            -0.3  0.79  Graduate  
2             1.4  1.74   Dropout  
3            -0.8 -3.12  Graduate  
4            -0.3  0.79  Graduate  

[5 rows x 37 columns]

In [137]:
#Checking number of Rows & Columns.
df.shape

(4424, 37)

In [138]:
df.columns.tolist()

['Marital status',
 'Application mode',
 'Application order',
 'Course',
 'Daytime/evening attendance\t',
 'Previous qualification',
 'Previous qualification (grade)',
 'Nacionality',
 "Mother's qualification",
 "Father's qualification",
 "Mother's occupation",
 "Father's occupation",
 'Admission grade',
 'Displaced',
 'Educational special needs',
 'Debtor',
 'Tuition fees up to date',
 'Gender',
 'Scholarship holder',
 'Age at enrollment',
 'International',
 'Curricular units 1st sem (credited)',
 'Curricular units 1st sem (enrolled)',
 'Curricular units 1st sem (evaluations)',
 'Curricular units 1st sem (approved)',
 'Curricular units 1st sem (grade)',
 'Curricular units 1st sem (without evaluations)',
 'Curricular units 2nd sem (credited)',
 'Curricular units 2nd sem (enrolled)',
 'Curricular units 2nd sem (evaluations)',
 'Curricular units 2nd sem (approved)',
 'Curricular units 2nd sem (grade)',
 'Curricular units 2nd sem (without evaluations)',
 'Unemployment rate',
 'Inflation r

In [139]:
#checking overall information of data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4424 entries, 0 to 4423
Data columns (total 37 columns):
 #   Column                                          Non-Null Count  Dtype  
---  ------                                          --------------  -----  
 0   Marital status                                  4424 non-null   int64  
 1   Application mode                                4424 non-null   int64  
 2   Application order                               4424 non-null   int64  
 3   Course                                          4424 non-null   int64  
 4   Daytime/evening attendance	                     4424 non-null   int64  
 5   Previous qualification                          4424 non-null   int64  
 6   Previous qualification (grade)                  4424 non-null   float64
 7   Nacionality                                     4424 non-null   int64  
 8   Mother's qualification                          4424 non-null   int64  
 9   Father's qualification                   

In [140]:
#Checking for null or missing values
df.isnull().sum()

Marital status                                    0
Application mode                                  0
Application order                                 0
Course                                            0
Daytime/evening attendance\t                      0
Previous qualification                            0
Previous qualification (grade)                    0
Nacionality                                       0
Mother's qualification                            0
Father's qualification                            0
Mother's occupation                               0
Father's occupation                               0
Admission grade                                   0
Displaced                                         0
Educational special needs                         0
Debtor                                            0
Tuition fees up to date                           0
Gender                                            0
Scholarship holder                                0
Age at enrollment                                 0
International                                     0
Curricular units 1st sem (credited)               0
Curricular units 1st sem (enrolled)               0
Curricular units 1st sem (evaluations)            0
Curricular units 1st sem (approved)               0
Curricular units 1st sem (grade)                  0
Curricular units 1st sem (without evaluations)    0
Curricular units 2nd sem (credited)               0
Curricular units 2nd sem (enrolled)               0
Curricular units 2nd sem (evaluations)            0
Curricular units 2nd sem (approved)               0
Curricular units 2nd sem (grade)                  0
Curricular units 2nd sem (without evaluations)    0
Unemployment rate                                 0
Inflation rate                                    0
GDP                                               0
Target                                            0
dtype: int64

In [148]:
#Describing Statistical data
df.describe()

Marital status       Course  Daytime/evening attendance\t  \
count     4424.000000  4424.000000                   4424.000000   
mean         1.178571  8856.642631                      0.890823   
std          0.605747  2063.566416                      0.311897   
min          1.000000    33.000000                      0.000000   
25%          1.000000  9085.000000                      1.000000   
50%          1.000000  9238.000000                      1.000000   
75%          1.000000  9556.000000                      1.000000   
max          6.000000  9991.000000                      1.000000   

       Previous qualification  Previous qualification (grade)  \
count             4424.000000                     4424.000000   
mean                 4.577758                      132.613314   
std                 10.216592                       13.188332   
min                  1.000000                       95.000000   
25%                  1.000000                      125.000000   
50%                  1.000000                      133.100000   
75%                  1.000000                      140.000000   
max                 43.000000                      190.000000   

       Admission grade    Displaced  Educational special needs       Debtor  \
count      4424.000000  4424.000000                4424.000000  4424.000000   
mean        126.978119     0.548373                   0.011528     0.113698   
std          14.482001     0.497711                   0.106760     0.317480   
min          95.000000     0.000000                   0.000000     0.000000   
25%         117.900000     0.000000                   0.000000     0.000000   
50%         126.100000     1.000000                   0.000000     0.000000   
75%         134.800000     1.000000                   0.000000     0.000000   
max         190.000000     1.000000                   1.000000     1.000000   

       Tuition fees up to date  ...  Curricular units 1st sem (approved)  \
count              4424.000000  ...                          4424.000000   
mean                  0.880651  ...                             4.706600   
std                   0.324235  ...                             3.094238   
min                   0.000000  ...                             0.000000   
25%                   1.000000  ...                             3.000000   
50%                   1.000000  ...                             5.000000   
75%                   1.000000  ...                             6.000000   
max                   1.000000  ...                            26.000000   

       Curricular units 1st sem (grade)  Curricular units 2nd sem (credited)  \
count                       4424.000000                          4424.000000   
mean                          10.640822                             0.541817   
std                            4.843663                             1.918546   
min                            0.000000                             0.000000   
25%                           11.000000                             0.000000   
50%                           12.285714                             0.000000   
75%                           13.400000                             0.000000   
max                           18.875000                            19.000000   

       Curricular units 2nd sem (enrolled)  \
count                          4424.000000   
mean                              6.232143   
std                               2.195951   
min                               0.000000   
25%                               5.000000   
50%                               6.000000   
75%                               7.000000   
max                              23.000000   

       Curricular units 2nd sem (evaluations)  \
count                             4424.000000   
mean                                 8.063291   
std                                  3.947951   
min                                  0.000000   
25%                           

In [149]:
#Getting Count of Target values
df["Target"].value_counts()

Target
Graduate    2209
Dropout     1421
Enrolled     794
Name: count, dtype: int64

In [143]:
df.head()

Marital status  Application mode  Application order  Course  \
0               1                17                  5     171   
1               1                15                  1    9254   
2               1                 1                  5    9070   
3               1                17                  2    9773   
4               2                39                  1    8014   

   Daytime/evening attendance\t  Previous qualification  \
0                             1                       1   
1                             1                       1   
2                             1                       1   
3                             1                       1   
4                             0                       1   

   Previous qualification (grade)  Nacionality  Mother's qualification  \
0                           122.0            1                      19   
1                           160.0            1                       1   
2                           122.0            1                      37   
3                           122.0            1                      38   
4                           100.0            1                      37   

   Father's qualification  ...  Curricular units 2nd sem (credited)  \
0                      12  ...                                    0   
1                       3  ...                                    0   
2                      37  ...                                    0   
3                      37  ...                                    0   
4                      38  ...                                    0   

   Curricular units 2nd sem (enrolled)  \
0                                    0   
1                                    6   
2                                    6   
3                                    6   
4                                    6   

   Curricular units 2nd sem (evaluations)  \
0                                       0   
1                                       6   
2                                       0   
3                                      10   
4                                       6   

   Curricular units 2nd sem (approved)  Curricular units 2nd sem (grade)  \
0                                    0                          0.000000   
1                                    6                         13.666667   
2                                    0                          0.000000   
3                                    5                         12.400000   
4                                    6                         13.000000   

   Curricular units 2nd sem (without evaluations)  Unemployment rate  \
0                                               0               10.8   
1                                               0               13.9   
2                                               0               10.8   
3                                               0                9.4   
4                                               0               13.9   

   Inflation rate   GDP    Target  
0             1.4  1.74   Dropout  
1            -0.3  0.79  Graduate  
2             1.4  1.74   Dropout  
3            -0.8 -3.12  Graduate  
4            -0.3  0.79  Graduate  

[5 rows x 37 columns]

In [144]:
#Dropping useless columns
drop_cols = ["Mother's qualification","Mother's occupation","Father's occupation","Father's qualification","Nacionality","Application mode","Application order","Curricular units 1st sem (without evaluations)","Curricular units 2nd sem (without evaluations)"]
df = df.drop(columns=drop_cols)

In [150]:
#Checking for unique values in each column
df.nunique()

Marital status                              6
Course                                     17
Daytime/evening attendance\t                2
Previous qualification                     17
Previous qualification (grade)            101
Admission grade                           620
Displaced                                   2
Educational special needs                   2
Debtor                                      2
Tuition fees up to date                     2
Gender                                      2
Scholarship holder                          2
Age at enrollment                          46
International                               2
Curricular units 1st sem (credited)        21
Curricular units 1st sem (enrolled)        23
Curricular units 1st sem (evaluations)     35
Curricular units 1st sem (approved)        23
Curricular units 1st sem (grade)          805
Curricular units 2nd sem (credited)        19
Curricular units 2nd sem (enrolled)        22
Curricular units 2nd sem (evaluations)     30
Curricular units 2nd sem (approved)        20
Curricular units 2nd sem (grade)          786
Unemployment rate                          10
Inflation rate                              9
GDP                                        10
Target                                      3
dtype: int64

In [151]:
df.head()

Marital status  Course  Daytime/evening attendance\t  \
0               1     171                             1   
1               1    9254                             1   
2               1    9070                             1   
3               1    9773                             1   
4               2    8014                             0   

   Previous qualification  Previous qualification (grade)  Admission grade  \
0                       1                           122.0            127.3   
1                       1                           160.0            142.5   
2                       1                           122.0            124.8   
3                       1                           122.0            119.6   
4                       1                           100.0            141.5   

   Displaced  Educational special needs  Debtor  Tuition fees up to date  ...  \
0          1                          0       0                        1  ...   
1          1                          0       0                        0  ...   
2          1                          0       0                        0  ...   
3          1                          0       0                        1  ...   
4          0                          0       0                        1  ...   

   Curricular units 1st sem (grade)  Curricular units 2nd sem (credited)  \
0                          0.000000                                    0   
1                         14.000000                                    0   
2                          0.000000                                    0   
3                         13.428571                                    0   
4                         12.333333                                    0   

   Curricular units 2nd sem (enrolled)  \
0                                    0   
1                                    6   
2                                    6   
3                                    6   
4                                    6   

   Curricular units 2nd sem (evaluations)  \
0                                       0   
1                                       6   
2                                       0   
3                                      10   
4                                       6   

   Curricular units 2nd sem (approved)  Curricular units 2nd sem (grade)  \
0                                    0                          0.000000   
1                                    6                         13.666667   
2                                    0                          0.000000   
3                                    5                         12.400000   
4                                    6                         13.000000   

   Unemployment rate  Inflation rate   GDP    Target  
0               10.8             1.4  1.74   Dropout  
1               13.9            -0.3  0.79  Graduate  
2               10.8             1.4  1.74   Dropout  
3                9.4            -0.8 -3.12  Graduate  
4               13.9            -0.3  0.79  Graduate  

[5 rows x 28 columns]

**COLUMN DESCRIPTION **

**1** **Marital Status**: 1 – single, 2 – married, 3 – widower, 4 – divorced, 5 – facto union, 6 – legally separated.


**2** **Course**: 33 - Biofuel Production Technologies, 171 - Animation and Multimedia Design, 8014 - Social Service, 9003 - Agronomy, 9070 - Communication Design, 9085 - Veterinary Nursing, 9119 - Informatics Engineering, 9130 - Equinculture, 9147 - Management, 9238 - Social Service, 9254 - Tourism, 9500 - Nursing, 9556 - Oral Hygiene, 9670 - Advertising and Marketing Management, 9773 - Journalism and Communication, 9853 - Basic Education, 9991 - Management.

**3** **Daytime/evening attendance**: 1 – daytime ,0 - evening.

**4** **Previous qualification**: 1 - Secondary education, 2 - Higher education - bachelor's degree, 3 - Higher education - degree, 4 - Higher education - master's, 5 - Higher education - doctorate, 6 - Frequency of higher education, 9 - 12th year of schooling - not completed, 10 - 11th year of schooling - not completed, 12 - Other - 11th year of schooling ,14 - 10th year of schooling ,15 - 10th year of schooling - not completed, 19 - Basic education 3rd cycle (9th/10th/11th year) or equiv, 38 - Basic education 2nd cycle (6th/7th/8th year) or equiv, 39 - Technological specialization course, 40 - Higher education - degree (1st cycle), 42 - Professional higher technical course, 43 - Higher education - master (2nd cycle).

**5** **Previous qualification (grade)** : Grade of previous qualification (between 0 and 200)

**6** **Admission grade**: Admission grade (between 0 and 200).

**7** **Displaced**: 	1 – yes, 0 – no.

**8** **Educational special needs**: 	1 – yes, 0 – no.

**9** **Debtor**: 1 – yes, 0 – no.

**10** **Tuition fees up to date**: 1 – yes, 0 – no.

**11** **Gender**: 1 – male, 0 – female

**12** **Scholarship holder**:  1 – yes, 0 – no.

**13** **International**:  1 – yes, 0 – no.

**14** **Curricular units 1st sem (grade)**: Grade average in the 1st semester (between 0 and 20)

**15** **Curricular units 2nd sem (grade)**: Grade average in the 2nd semester (between 0 and 20)